src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php

```diff
@@ -0,0 +1,779 @@
+                $aClass .= ' activeConnectedNode';
+                $treeNodeDataModel->setOpened(true);
+                $treeNodeDataModel->setSelected(true);
+
+                if ($this->primaryConnectedNodeIdOfCurrentPage === $node->id) {
+                    $aClass .= ' primaryConnectedNode';
+                    $treeNodeDataModel->setDisabled(true);
+                }
+            } else {
+                $aClass .= ' otherConnectedNode';
+            }
+        }
+
+        $liAttr = array_merge($liAttr, ['class' => $liClass]);
+        $treeNodeDataModel->setLiAttr($liAttr);
+
+        $aAttr = array_merge($aAttr, ['class' => $aClass]);
+        $treeNodeDataModel->setAAttr($aAttr);
+    }
+
+    private function setInitialType(\TdbCmsTree $node): string
+    {
+        $children = $node->GetChildren(true);
+        if ($children->Length() > 0) {
+            $type = 'folder';
+        } else {
+            $type = 'page';
+        }
+
+        if ('' !== $node->sqlData['link']) {
+            $type = 'externalLink';
+        }
+
+        if (in_array($node->id, $this->restrictedNodes)) {
+            $type .= 'RestrictedMenu';
+        }
+
+        if (true == $node->fieldHidden) {
+            $type = 'node-hidden';
+        }
+
+        return $type;
+    }
+
+    /**
+     * Fetches a list of all restricted nodes (of all portals)
+     * a restricted node is a startnavigation nodes of a portal.
+     *
+     * @return array
+     */
+    private function getPortalNavigationStartNodes(): array
+    {
+        $portalList = \TdbCmsPortalList::GetList(null, \TdbCmsUser::GetActiveUser()->GetCurrentEditLanguageID());
+
+        $restrictedNodes = array();
+        while ($portal = $portalList->Next()) {
+            $restrictedNodes[] = $portal->fieldMainNodeTree;
+            $navigationList = $portal->GetFieldPropertyNavigationsList();
+            while ($navigation = $navigationList->Next()) {
+                $restrictedNodes[] = $navigation->fieldTreeNode;
+            }
+        }
+
+        return $restrictedNodes;
+    }
+
+    /**
+     * Get the sort field name for the active language.
+     * Get default field name if active language is same as portal main language,
+     * field is not translatable or field based translation is off.
+     *
+     * @return string
+     */
+    private function getSortFieldName(): string
+    {
+        $entrySortField = 'entry_sort';
+        if (\TdbCmsTree::CMSFieldIsTranslated('entry_sort')) {
+            $sLanguagePrefix = \TGlobal::GetLanguagePrefix($user = \TdbCmsUser::GetActiveUser()->GetCurrentEditLanguageID());
+            if ('' !== $sLanguagePrefix) {
+                $entrySortField .= '__'.$sLanguagePrefix;
+            }
+        }
+
+        return $entrySortField;
+    }
+
+    /**
+     * @param bool $chachingEnabled
+     */
+    private function setCaching($cachingEnabled, \IMapperCacheTriggerRestricted $cacheTriggerManager) {
+        if (true === $cachingEnabled) {
```

Comment:
change to if (false === $achingEnabled) return;

Reply:
ok